src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Container.cs

```diff
@@ -265,60 +225,67 @@ protected virtual void ValidateName(IComponent component, string name)
                 throw new ArgumentNullException(nameof(component));
             }
 
-            if (name != null)
+            if (name == null)
             {
-                for (int i = 0; i < Math.Min(_siteCount, _sites.Length); i++)
-                {
-                    ISite s = _sites[i];
+                return;
+            }
```

Comment:
Early-out like this helps reduce nesting, which can make code easier to reason about. I guess there is not consensus about that though.